toolkit/tools/pkg/licensecheck/licensecheck.go

```diff
@@ -0,0 +1,411 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/*
+Package licensecheck provides a tool for searching RPMs for bad licenses, as well as several directly callable functions.
+The core of the tool is the LicenseChecker struct, which is responsible for searching RPMs for bad licenses. The tool is
+based on a 'simpletoolchroot' which is a wrapper that allows for easy chroot creation and cleanup.
+
+The lifecycle of the LicenseChecker is as follows:
+
+1. Create a new LicenseChecker with New()
+
+2. Call CheckLicenses() to search for bad licenses
+
+3. Either:
+  - Call FormatResults() to get a formatted string of the results
+  - Call GetAllResults() to get all the results, split into buckets.
+
+4. Call CleanUp() to tear down the chroot
+
+Also provided are several directly callable functions (these expect to be run in an environment with the necessary
+macros, i.e. a chroot): CheckRpmLicenses(), IsALicenseFile(), IsASkippedLicenseFile()
+
+The LicenseCheckerResult struct is used to store the results of the search. It contains the path to the RPM, a list of
+bad documents, a list of bad files, and a list of duplicated documents. The bad documents are %doc files that are not
+at least also in the license files. The bad files are general files that are misplaced in the licenses directory.
+
+The duplicated documents are %doc files that are also in the license files. These are not technically bad, but are messy
+and should be cleaned up.
+*/
+package licensecheck
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/rpm"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/sliceutils"
+	"github.com/microsoft/azurelinux/toolkit/tools/pkg/simpletoolchroot"
+)
+
+const (
+	chrootName      = "license_chroot"
+	licensePrefix   = "/usr/share/licenses"
+	numParallelJobs = 20
```

Comment:
The `chrootName` and `numParallelJobs` are used only inside the same one function. No need to make them globals in such case and keeping them closer to the code would also make it easier to read.